scripts/o3de/o3de/engine_template.py

```diff
@@ -2103,15 +2103,29 @@ def create_gem(gem_path: pathlib.Path,
 
     if summary:
         replacements.append(("${Summary}", summary))
+    else:
+        replacements.append(("${Summary}", ""))
+    
     if license:
         replacements.append(("${License}", license))
+    else:
+        replacements.append(("${License}", ""))
+    
     if license_url:
         replacements.append(("${LicenseURL}", license_url))
+    else:
+        replacements.append(("${LicenseURL}", ""))
+    
     if origin:
         replacements.append(("${Origin}", origin))
+    else:
+        replacements.append(("${Origin}", ""))
+    
     if origin_url:
         replacements.append(("${OriginURL}", origin_url))
-    
+    else:
+        replacements.append(("${OriginURL}", ""))
+
```

Comment:
If we do this won't it remove all the helper text in each gem templates' gem.json or is that the goal?

Reply:
It shouldn't affect the helper text, since only the `${VAR_NAME}` pattern is affected in these string replacements. Any pattern that matches `${VAR_NAME}` will be affected in the destination gem.json file (the file that the create/edit gem workflow creates) however. 

If you do not add these empty string replacements, the destination file ends up looking like this (actual output of a test gem I made from the workflow):
```
{
    "gem_name": "kevingem6",
    "display_name": "6th gem of kevin",
    "license": "mit",
    "license_url": "${LicenseURL}",
    "origin": "tkothadev",
    "origin_url": "${OriginURL}",
    "type": "Code",
    "summary": "${Summary}",
    "canonical_tags": [
        "Gem"
    ],
    "user_tags": [
        "kevingem6"
    ],
    "icon_path": ".",
    "requirements": "${Requirements}",
    "documentation_url": "${DocumentationURL}",
    "dependencies": [],
    "repo_uri": "${RepoURI}",
    "restricted": "kevingem6"
}
```

In practice, dealing with this in the edit workflow was extremely annoying. Especially the last page when I go to 'Edit' the gem, it would keep autofilling the field with `${RepoURI}`, which should not exist as a value in the workflow. This would have to be cleared out everytime before I proceed. The `RepoURI` is a non-trivial example, since I am simulating creating a local gem in which I have no plans to host in a repository.

Reply:
OK, thanks for the clear explanation!